message_parser/src/test/java/cumulus_sled/message_parser/MessageParserTest.java

```diff
@@ -72,6 +72,39 @@ public void testMessageAdapter()
         }
     }
 
+    /*
+     * Test that the message handler is hitting all of the correct functions and converting the params
+     * to JSON correctly if an alternate path is specified
+     */
+    @Test
+    public void testMessageAdapterAlternate()
+    {
+        class MessageAdapterMock extends MessageAdapter {
```

Comment:
If there's a better way to do this without bringing in a full mock lib , I'm open to it.

Reply:
Could we set the `CUMULUS_MESSAGE_ADAPTER_DIR` environment variable instead of a mock lib? How are we really verifying it actually used the alternate path?

Reply:
Sadly, no, we can't, due to the way the JVM handles environment variables.   We could, in theory script something, but that seems like overkill versus subclassing like this.

Reply:
> How are we really verifying it actually used the alternate path?

Not implicitly.  I'm not a huge fan of this, but short of downloading a custom CMA and having a different response, which feels fragile....    the CMA doesn't implicitly identify it's versioning.   This test is really to show that if GetMessageAdapterEnvironmentVariable is set to a known directory it doesn't break.      I suppose we could force it to run last and remove the default CMA as a way to explicitly test it's not hitting the default.  

🤔

Reply:
Actually a separate test class should sort this.

Reply:
@laurenfrederick  How about: 69d936b ?   It's not entirely as elegant as I'd like, but it addresses your concern, I think.